lib/spree/core/calculated_adjustments.rb

```diff
@@ -43,7 +43,8 @@ def create_adjustment(label, target, calculable, mandatory = false, state = "clo
               order: order_object_for(target),
               label: label,
               mandatory: mandatory,
-              state: state
+              state: state,
+              included: tax_included?(self, target)
```

Comment:
Hm, where else are adjustments created?

- `app/helpers/checkout_helper.rb` creates in-memory adjustments. That's probably fine. Not sure about all use cases.
- `app/models/spree/return_authorization.rb` creates a credit which is not tax and not included in the price. All good.
- `app/models/spree/tax_rate.rb` creates tax adjustments and doesn't set the `included` attribute. :warning: 

Should we add the attribute in that last case?

Reply:
Ah yes... well there are two places in TaxRate which call `#create_adjustment`, which comes back to `Spree::CalculatedAdjustments#create_adjustment` :heavy_check_mark: 

There's one place that doesn't, and it's for a very specific type of tax refund, and think in that case they are supposed to be marked as not inclusive (which is the default at database level, if nothing is specified). The other thing about that specific tax refund case, is that I don't think any instances currently have tax settings that would actually ever lead them to reach that section of code. The condition is: instances using inclusive tax (only EU and AUS) that _also_ apply different tax rates in different tax zones (only US and CA). Nobody is doing both at the same time. The logic around that code is completely rewritten in Spree 2.2, but I haven't got to that bit yet. I think the current version is basically not correct, so it's a good thing we never actually use it! :sweat_smile: